superset-frontend/src/hooks/apiResources/sqlEditorTabs.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { pickBy } from 'lodash';
+import { QueryEditor } from 'src/SqlLab/types';
+import { api, JsonResponse } from './queryApi';
+
+export type EditorMutationParams = {
+  queryEditor: QueryEditor;
+  extra?: Record<string, any>;
+};
+
+const sqlEditorApi = api.injectEndpoints({
+  endpoints: builder => ({
+    updateSqlEditorTab: builder.mutation<JsonResponse, EditorMutationParams>({
+      query: ({
+        queryEditor: {
```

Comment:
Given that the query editor state is saved as a JSON object, it would be a good idea to add a version field to the object. If we have schema changes in the future, we'll be able to create transformers between the different versions and preserve compatibility. This is one of the main problems we face today with other JSON objects.